### DISCUSSION TITLE
AuthN and AuthZ

### DISCUSSION THREAD

**Discussion by gustavogq** [hanamimastery/episodes#10] [Suggestions]
Hi, my suggestion, as someone starting with Hanami, is how is the best way to work with authentication and authorization. Mostly of the gems are too focused on Rails apps, and I'm thinking kind of difficult to do authN and authZ in Hanami2

**Comment by swilgosz** [comment]
Thanks! 

I already have an episode about authorization with JWT, and I will tackle AuthN for sure in the upcoming months :)

https://hanamimastery.com/episodes/12-authorization-with-jwt

When Hanami 2.1 comes out, I will prepare integration with Rodauth showcase.

Will that work?

Already, for AuthN rodauth is my goto at the moment

**Reply by gustavogq** [reply]
Thank you for the reply! I will take a look in the episode and at Rodauth!
